src/metabase/api/model_index.clj

```diff
@@ -0,0 +1,81 @@
+(ns metabase.api.model-index
+  (:require
+   [compojure.core :refer [POST]]
+   [metabase.api.common :as api]
+   [metabase.models.card :refer [Card]]
+   [metabase.models.model-index :as model-index :refer [ModelIndex]]
+   [metabase.task.index-values :as task.index-values]
+   [metabase.util.i18n :refer [tru]]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(defn- ensure-type
+  "Ensure that the ref exists and is of type required for indexing."
+  [t ref metadata]
+  (if-let [field (some (fn [f] (when ((comp #{(model-index/normalize-field-ref ref)} :field_ref) f)
+                                 f))
+                       metadata)]
+    (let [type-slot (case t
+                      :type/PK :semantic_type
+                      :type/Text :effective_type)]
+      (when-not (isa? (type-slot field) t)
+        (throw (ex-info (tru "Field is not of type {0}" t)
+                        {:status-code   400
+                         :expected-type t
+                         :type          (:effective_type field)
+                         :field         (:name field)}))))
+    (throw (ex-info (tru "Could not identify field by ref {0}" ref)
+                    {:status-code 400
+                     :ref         ref
+                     :fields      metadata}))))
+
+(api/defendpoint POST "/"
+  [:as {{:keys [model_id pk_ref value_ref] :as _model-index} :body}]
+  {model_id  ms/PositiveInt
+   pk_ref    any?
+   value_ref any?}
+  (let [model      (api/read-check Card model_id)
```

Comment:
```suggestion
  (api/check-superuser)
  (let [model      (api/read-check Card model_id)
```

Per the engineering sketch, we should limit this to admins only. We also need this for `DELETE`